src/square.py

```diff
@@ -112,14 +130,3 @@ def __repr__(self):
     def __add__(self, other: Vector):
         """Add vector (displacement) to a square."""
         return Square(super().__add__(other))
-
-    def is_in_board(self) -> bool:
-        """Agnostic conditions that apply to all squares.
-
-        In particular this contains boundary checks, since they are decipherable from square notation basically.
-        NOTE: This could go to `Board` instead too, I'm just trying to semantically allocate conditioning across objects.
-
-        Returns:
-            If square respects conditions.
-        """
-        return 0 <= self.rank < 8 and 0 <= self.file < 8
```

Comment:
I rendered this method obsolete my making `Square` objects self-constrained. See [`__new__` method](https://github.com/TrentHawkins/Chess/pull/17/commits/7887e1699eef7117f73a6d58f8895cc2bea97208#r960090071).